File: images/toolchain/alpine-3.24/Dockerfile.scala
# Toolchain image: Scala (scala-cli + sbt), Alpine/musl.
ARG BASE_IMAGE=toolchain-java:alpine-3.24
FROM ${BASE_IMAGE}
COPY cache/scala-cli-x86_64-pc-linux-static.gz /tmp/scala-cli.gz
RUN gunzip -c /tmp/scala-cli.gz > /usr/local/bin/scala-cli && chmod +x /usr/local/bin/scala-cli && rm /tmp/scala-cli.gz
COPY cache/sbt-1.13.0.tgz /tmp/sbt.tgz
RUN tar -xzf /tmp/sbt.tgz -C /opt && rm /tmp/sbt.tgz
ENV PATH=/opt/sbt/bin:$PATH
RUN scala-cli version && sbt --script-version

File: images/toolchain/alpine-3.24/Dockerfile.php
# Toolchain image: PHP + Composer, Alpine.
#
# static-php-cli publishes a statically linked PHP CLI (no libc dependency), so
# it runs on musl as-is. Both artifacts are pre-downloaded (fetch-artifacts.sh)
# and COPYed.
ARG BASE_IMAGE=toolchain-base:alpine-3.24
FROM ${BASE_IMAGE}
COPY cache/php-8.5.8-cli-linux-x86_64.tar.gz /opt/php.tar.gz
COPY cache/composer.phar /opt/composer.phar
RUN mkdir -p /opt/php && tar -xzf /opt/php.tar.gz -C /opt/php && rm /opt/php.tar.gz \
 && mkdir -p /opt/php/bin 2>/dev/null || true
RUN printf '#!/bin/sh\nexec /opt/php/php /opt/composer.phar "$@"\n' > /usr/local/bin/composer \
 && chmod +x /usr/local/bin/composer
ENV PATH=/opt/php:$PATH
